R/DataPackage.R

```diff
@@ -1588,161 +1571,162 @@ setMethod("serializeToBagIt", signature("DataPackage"), function(x, mapId=NA_cha
-        relFile <- paste(relFile, dataObj@filename, sep = "")
-    } else {
-        # If the filename wasn't specified, use the identifier
-        relFile <- paste(relFile, getIdentifier(dataObj), sep="")
+    # Create a temp working area where the BagIt directory structure will be created
+    tmpDir <- tempdir()
+    bagDir <- file.path(tmpDir, "bag")
+    if(file.exists(bagDir)) {
+        unlink(bagDir, recursive=TRUE)
     }
+    # Create the bag directories
+    dir.create(bagDir)
+    payloadDir <- file.path(bagDir, "data")
+    if(!file.exists(payloadDir)) dir.create(payloadDir)
+    metadataDir <- file.path(bagDir, "metadata")
+    if(!file.exists(metadataDir)) dir.create(metadataDir)
+    sysMetaDir <- file.path(metadataDir, "sysmeta")
+    if(!file.exists(sysMetaDir)) dir.create(sysMetaDir)
     
-    # Create the directory if it doesn't exist
-    if(!file.exists(dirname(relFile))) {
-        dir.create(dirname(relFile))
+    # Create bagit.txt
+    bagitFileText <- sprintf("BagIt-Version: 0.97\nTag-File-Character-Encoding: UTF-8")
+    writeLines(bagitFileText, file.path(bagDir, "bagit.txt"))
+
+    # Create a ResourceMap object and serialize it
+    if(is.na(mapId)) {
+        mapId <- sprintf("urn:uuid:%s", uuid::UUIDgenerate())
+    }
+    if(is.na(syntaxName)) {
+        syntaxName="rdfxml"
+    }
+    if(is.na(mimeType)) {
+        mimeType <- "application/rdf+xml"
     }
+    if(is.na(resolveURI)) {
+        resolveURI <- ""
+    }
+    tmpFile <- tempfile()
+    serializePackage(x, file=tmpFile, id=mapId, syntaxName=syntaxName, namespaces=namespaces,
+                     mimeType=mimeType, resolveURI=resolveURI, creator=creator)
     
-    if(!is.na(dataObj@filename)) {
-      if(file.exists(dataObj@filename)) {
-        file.copy(dataObj@filename, relFile)
-        # Add this data pacakge member to the bagit files
-        pidMap <- c(pidMap, sprintf("%s %s", identifiers[idNum], relFile))
-        thisMd5 <- digest(relFile, algo="md5", file=TRUE)
-        manifest <- c(manifest, sprintf("%s %s", as.character(thisMd5), relFile))  
-      } else {
-        stop(sprintf("Error serializing to BagIt format, data object \"%s\", uses file %s but this file doesn't exist", dataObj@filename, identifiers[idNum]))
-      }
+    relFile <- file.path("oai-ore.xml")
+    resMapFilepath <- file.path(metadataDir, relFile)
+    file.copy(tmpFile, resMapFilepath)
+    # Add resource map to the manifrest
+    resMapMd5 <- digest(resMapFilepath, algo="md5", file=TRUE)
+    manifestFileTextLine <- sprintf("%s %s", resMapMd5, gsub(paste(".*bag", .Platform$file.sep, sep=""), "", resMapFilepath))
+    writeLines(manifestFileTextLine, file.path(bagDir, "tagmanifest-md5.txt"))
+
+    identifiers <- getIdentifiers(x)
+    # Get a resource map so that it can be queried for science metadata
+    resMap <- getResourceMap(x)
+    # Check to see if the user supplied any science metadata
+    scienceMetadataUris <- get_science_metadata(resMap)
```

Comment:
Use slightly more verbose function name
```suggestion
    science_metadata_uris <- get_science_metadata_uris(resMap)
```